File: jewels-and-stones/jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        //check for null input 
        if(jewels == null || stones == null)return 0;
        int res =0;
        HashSet<Character> jewelSet = new HashSet<>();
        for(char j: jewels.toCharArray()) jewelSet.add(j);
        for(char s: stones.toCharArray()){
            if(jewelSet.contains(s)) res++;
        }
        return res;
    }
}
//Time complexity:Max(J, S)
//space complexity: J

File: jewels-and-stones/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        #check for null input 
        if not jewels or not stones: return 0
        jewelSet = set()
        res = 0
        for j in jewels: jewelSet.add(j)
        for s in stones: 
            if s in jewelSet: res+=1
        return res